### PR TITLE
Addition of global_throttle flag to variable hierarchy

### LINKED ISSUE #554: Add flag for local/global throttle normalization
### Desired capability or behavior.

There is a hidden (i.e. hardcoded) option to toggle which normalization scheme is used by `engine_deck.py` on line 185-186. These should be exposed as an option that can be passed/defined when setting up an Aviary (`Aircraft.Engine.GLOBAL_NORMALIZATION` or something similar?).

Some discussion on defaults might be needed - local normalization is pretty much universally desired except when replicating results for GASP-translated engine decks, which need global right now. The fix is to add a T4 constraint to each individual mission segment to set throttle to matche what the engine ran at when used in GASP, but that may be out of scope for this version update. If we don't do that, then 2DOF benchmark tests may provide slightly different values, which is ok, just may not match quite as closely to legacy results.

If we do local normalization as a default, then a new test that ensures the normalization worked properly is required (with a little dummy data set designed solely to test this feature)

### Is your feature request related to a problem? Please describe.

_No response_

### Associated Bug Report

_No response_

### PR DESCRIPTION
**Pull request by jkirk5**
### Summary

Exposes the previously hidden option to toggle if EngineDecks used a global throttle range across the entire data set, or disable it to normalize the throttle per flight condition (the default).

GASP-converted input files have global_throttle set to True, as GASP-based engine decks have varied throttle ranges and do not produce matching results if throttle is normalized per flight condition.

Will route this to 0.9.6.dev branch once reviews are in and it is still passing testing

### Related Issues

- Resolves #554 

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/subsystems/mass/gasp_based/test/test_mass_summation.py
Comment (ehariton) [inline aviary/subsystems/mass/gasp_based/test/test_mass_summation.py:1]: Did you mean to commit these changes in this PR?

File: aviary/subsystems/propulsion/engine_deck.py [hunk 1]
```diff
@@ -180,12 +184,20 @@ def __init__(self, name='engine_deck', options: AviaryValues = None,
         # Create dict for variables present in engine data with associated units
         self.engine_variables = {}
 
-        # TODO make this an option - disabling global throttle ranges is better to
-        #      prevent unintended extrapolation, but breaks missions using GASP-based
-        #      engines that have uneven throttle ranges (need t4 constraint on mission
-        #      to truly fix)
-        self.global_throttle = True
-        self.global_hybrid_throttle = True
+        if Aircraft.Engine.GLOBAL_THROTTLE in options:
```
Comment (ehariton) [inline aviary/subsystems/propulsion/engine_deck.py:187]: So it looks like these options are exclusive, they could both be true or both be false?

**Review by ehariton — APPROVED** [review]

File: aviary/subsystems/mass/gasp_based/test/test_mass_summation.py
Comment (jkirk5) [reply aviary/subsystems/mass/gasp_based/test/test_mass_summation.py:1]: Looks like I was debugging this test at some point

File: aviary/subsystems/mass/gasp_based/test/test_mass_summation.py [hunk 2]
```diff
@@ -3318,4 +3318,7 @@ def test_case1(self):
 
 
 if __name__ == "__main__":
-    unittest.main()
+    # unittest.main()
+    test = MassSummationTestCase1()
+    test.setUp()
+    test.test_case1()
```
Comment (jkirk5) [inline aviary/subsystems/mass/gasp_based/test/test_mass_summation.py:3324]:
```suggestion
    unittest.main()
```

File: aviary/subsystems/propulsion/engine_deck.py (hunk 1 above)
Comment (jkirk5) [reply aviary/subsystems/propulsion/engine_deck.py:187]: Yeah you can have different values for hybrid vs. regular throttle

**Pull request merged**